docs/en/wallet-attestation.rst

```diff
@@ -30,271 +33,271 @@ The following requirements for the Wallet Attestation are met:
-||       || App Attestation Service.                                   |
-+--------+-------------------------------------------------------------+
-|| cnf   || JSON object, according to                                  |
-||       || `RFC7800 <https://www.rfc-editor.org/rfc/rfc7800.html>`_   |
-||       || containing the public part of an asymmetric key pair owned |
-||       || by the Wallet Instance.                                    |
-+--------+-------------------------------------------------------------+
-|| iat   || Unix timestamp of attestation request                      |
-||       || issuance time.                                             |
-+--------+-------------------------------------------------------------+
-|| exp   || Unix timestamp regarding the                               |
-||       || expiration date time.                                      |
-+--------+-------------------------------------------------------------+
-
-
-Below a non-normative example of the Wallet Attestation
-request where the decoded JWS headers and payload are separated by a comma:
-
-.. code-block:: javascript
+    GET /nonce HTTP/1.1
+    Host: walletprovider.example.com
+
+.. code-block:: http
+
+    HTTP/1.1 200 OK
+    Content-Type: application/json
+
+    {
+      "nonce": "d2JhY2NhbG91cmVqdWFuZGFt"
+    }
+
+**Step 7**: The Wallet Instance:
+
+  * generate a ``client_data`` which is a JSON containing ``challenge`` and ephemeral public ``jwk``.
+  * generate a ``client_data_hash`` as SHA256 of ``client_data``.
+
+Below a non-normative example of the ``client_data``.
+
+.. code-block:: json
+
+  {
+    "challenge": "0fe3cbe0-646d-44b5-8808-917dd5391bd9",
+    "jwk": {
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "4HNptI-xr2pjyRJKGMnz4WmdnQD_uJSq4R95Nj98b44",
+        "y": "LIZnSB39vFJhYgS3k7jXE4r3-CoGFQwZtPBIRqpNlrg",
+        "kid": "vbeXJksM45xphtANnCiG6mCyuU4jfGNzopGuKvogg9c"
+      }
+  }
+
+**Steps 8-10**: The Wallet Instance:
+
+  * generate an ``hardware_signature`` signing ``client_data_hash`` with Wallet Hardware private key. It will be used as a **PoP** of the Wallet Hardware Keys.
+  * ask to Device Integrity Service to generate an ``integrity_assertion`` bounded with ``client_data_hash``.
+  * obtain from Device Integrity Service an ``integrity_assertion`` signed by OEM.
+
+.. note:: ``integrity_assertion`` is a custom payload generated by Device Integrity Service, signed by device OEM and encoded in base64 to have uniformity between different devices.
+
+**Steps 11-12**: The Wallet Instance:
+  * generate the Wallet Attestation Request as a JWT containing ``integrity_assertion``, ``hardware_signature``, ``challenge``, ``wallet_hardware_key_tag``, ``public_jwk`` signed with the private key associated with the public key of the ephemeral key pair generated initially.
+  * send the Wallet Attestation Request to Wallet Provider backend via token endpoint.
```

Comment:
```suggestion
**Steps 11-12**: The Wallet Instance carries out the following actions:
  *  Constructs the Wallet Attestation Request in the form of a JWT. This JWT includes the ``integrity_assertion``, ``hardware_signature``, ``challenge``, ``wallet_hardware_key_tag``, and ``public_jwk``, and is signed using the private key from the initially generated ephemeral key pair.
  *  Submits the Wallet Attestation Request to the Wallet Provider's backend through the token endpoint.
```

Reply:
I assume that the token endpoint requires a specialized grant type for this specific flow/issuance

Reply:
the token endpoint issues different token types and detect the requested token type using the grant type. We have used a generic grant type, I'm wondering if we should specialize a grant type for this.

I would not handle this in this PR, but move this analysis starting from a new gh issue